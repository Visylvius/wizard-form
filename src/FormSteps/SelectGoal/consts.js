const optionsArr = [
  { value: 'footwear', displayValue: 'Footwear' },
  { value: 'outerware', displayValue: 'Outerware' },
  { value: 'bags', displayValue: 'Bags' },
  { value: 'other', displayValue: 'Other' },
];

const SELECT_GOAL = 'selectGoal';

export { SELECT_GOAL, optionsArr };
