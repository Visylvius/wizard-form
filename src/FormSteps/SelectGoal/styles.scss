.SelectGoal-Button {
  margin-top: 10px;
}
