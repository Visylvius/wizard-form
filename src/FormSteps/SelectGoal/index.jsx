import React from 'react';

import Select from '../../Components/Select';
import Button from '../../Components/Button';
import { DEFAULT_VALUE } from '../../globalConsts';

import { SELECT_GOAL, optionsArr } from './consts';
import './styles.scss';

const SelectGoal = ({
  currentStep,
  selectedVal,
  onChange,
  onButtonClick,
  testId,
}) => {
  if (currentStep !== 3) {
    return null;
  }

  return (
    <>
      <Select
        selectTestId={testId}
        label='What category would you like to leave feedback about?'
        selectedVal={selectedVal}
        name={SELECT_GOAL}
        onChange={onChange}
        optionsArr={optionsArr}
      />
      <Button
        data-testid='SelectGoal-Button'
        buttonClasses='SelectGoal-Button'
        disabled={selectedVal === DEFAULT_VALUE}
        onClick={onButtonClick}
        shouldCenter
      >
        Next
      </Button>
    </>
  );
};

export default SelectGoal;
