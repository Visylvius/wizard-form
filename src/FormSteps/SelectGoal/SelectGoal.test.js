import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import SelectGoal from './index';

describe('SelectGoal (Unit)', () => {
  const mockOnChange = jest.fn();
  const mockOnClick = jest.fn();
  const renderSelectGoal = props =>
    render(
      <SelectGoal
        testId='SelectGoal-select'
        currentStep={3}
        onChange={mockOnChange}
        onButtonClick={mockOnClick}
        selectedVal='default'
        {...props}
      />,
    );

  describe('render', () => {
    test('should render properly', () => {
      const tree = renderer.create(
        <SelectGoal
          currentStep={3}
          onChange={mockOnChange}
          onButtonClick={mockOnClick}
          selectedVal='default'
        />,
      );

      expect(tree).toMatchSnapshot();
    });

    test('should render null if the step is not 5', () => {
      const tree = renderer.create(
        <SelectGoal
          currentStep={2}
          onChange={mockOnChange}
          onButtonClick={mockOnClick}
          selectedVal='default'
        />,
      );

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Intergration', () => {
    test('should properly call onChange', async () => {
      const { getByTestId, getByText } = renderSelectGoal();

      fireEvent.change(getByTestId('SelectGoal-select'), {
        target: { value: 'other' },
      });

      await waitFor(() => expect(mockOnChange).toHaveBeenCalled());
      expect(getByText('Other')).toBeInTheDocument();
    });

    test('should properly call onClick', async () => {
      const { getByTestId } = renderSelectGoal({ selectedVal: 'outerware' });

      userEvent.click(getByTestId('SelectGoal-Button'));

      await waitFor(() => expect(mockOnClick).toHaveBeenCalled());
    });

    test('should not call onClick when the button is disabled', async () => {
      const { getByTestId } = renderSelectGoal();

      userEvent.click(getByTestId('SelectGoal-Button'));

      await waitFor(() => expect(mockOnClick).not.toHaveBeenCalled());
    });
  });
});
