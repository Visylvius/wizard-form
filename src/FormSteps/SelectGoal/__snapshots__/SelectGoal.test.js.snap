// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectGoal (Unit) render should render null if the step is not 5 1`] = `null`;

exports[`SelectGoal (Unit) render should render properly 1`] = `
Array [
  <label
    className="Select-label"
  >
    What category would you like to leave feedback about?
  </label>,
  <select
    className="Select-mainWrapper"
    name="selectGoal"
    onChange={[MockFunction]}
    value="default"
  >
    <option
      defaultValue={true}
      disabled={true}
      value="default"
    >
      Choose...
    </option>
    <option
      data-testid="SelectOption-0"
      value="footwear"
    >
      Footwear
    </option>
    <option
      data-testid="SelectOption-1"
      value="outerware"
    >
      Outerware
    </option>
    <option
      data-testid="SelectOption-2"
      value="bags"
    >
      Bags
    </option>
    <option
      data-testid="SelectOption-3"
      value="other"
    >
      Other
    </option>
  </select>,
  <div
    className="Button-container"
  >
    <button
      className="Button-component SelectGoal-Button"
      data-testid="SelectGoal-Button"
      disabled={true}
      onClick={[MockFunction]}
    >
      Next
    </button>
  </div>,
]
`;
